src/transformers/training_args.py

```diff
@@ -663,8 +663,19 @@ def __post_init__(self):
         self.lr_scheduler_type = SchedulerType(self.lr_scheduler_type)
         if self.do_eval is False and self.evaluation_strategy != IntervalStrategy.NO:
             self.do_eval = True
-        if self.eval_steps is None:
-            self.eval_steps = self.logging_steps
+
+        # eval_steps has to be defined and non-zero, fallbacks to logging_steps if the latter is non-zero
+        if self.do_eval and (self.eval_steps is None or self.eval_steps == 0):
+            if self.logging_steps > 0:
+                self.eval_steps = self.logging_steps
+            else:
+                raise ValueError(
+                    f"evaluation strategy {self.evaluation_strategy} requires either non-zero --eval_steps or --logging_steps"
+                )
+
+        # logging_steps must be non-zero for logging_strategy that is other than 'no'
+        if self.logging_strategy != IntervalStrategy.NO and self.logging_steps == 0:
```

Comment:
```suggestion
        if self.logging_strategy == IntervalStrategy.STEPS and self.logging_steps == 0:
```
The `EPOCH` logging strategy will not look at `logging_steps`, so no need to raise in that case.